lib/storage.js

```diff
@@ -459,9 +459,10 @@ api.remove = async ({edvId, id}) => {
   assert.string(id, 'id');
   _assert128BitId(id);
 
-  const result = await database.collections.edvDoc.deleteOne(
-    {edvId: database.hash(edvId), id: database.hash(id)});
-  return result.result.n !== 0;
+  const result = await database.collections.edvDoc.findOneAndUpdate(
+    {edvId: database.hash(edvId), id: database.hash(id)},
+    {$set: {'doc.indexed': [], 'doc.deleted': true, 'doc.jwe.ciphertext': ''}});
```

Comment:
Should this be using `$unset` with `doc.jwe.ciphertext`? Won't this just set `ciphertext` to an empty string? Maybe we want to clear out the `jwe` entirely. Is there some reasoning behind keeping other jwe fields (like `recipients`) around? (Presumably this approach does that).

Reply:
In our previous conversation on IRC, you had stated that `id`, `sequence`, and `jwe` should remain. If `ciphertext` is removed completely instead of being an empty string there would need to be additional checks for that in the edv-client before passing it to cipher to decrypt. 
This approach does keep all other `jwe` fields. It was my understanding from the use case in the issue that the recipients array was present.

Reply:
Hmm, well, if we're keeping `jwe`, I think it makes sense to require the client to clear the document contents before it is encrypted, i.e., it's not the server's responsibility to deal with the JWE in any way. Clearing only the `ciphertext` as we do here is problematic for a number of reasons and breaks expectations around using JWEs in general. So what remains is to decide what to do about this `deleted` flag.

Suppose we could make deletion a "client only" thing. It's up to the client to clear a document and set its `meta.deleted` flag to `true`. Then the client uses the `update` call like usual. Then the server knows nothing about deletions at all; the only question is: does the server need to know? I don't think the SDS WG (Secure Data Storage Working Group) has decided this yet.

Given that this is up in the air, the simplest thing to do, I think, would be to make deletion a client-side only affair. We can just remove the HTTP API for deleting from the server entirely -- and change the remove API in the client to do an update where it clears the content and sets a meta tag. What do you think about that?

Reply:
Agreed! Should I keep it on this PR or create a new branch since it's a different direction?

Reply:
@mandyvenables -- Hmm, I recommend keeping this PR because it's got all these comments on it. You can squash the history once you've made the appropriate other changes to clean it up though.

Reply:
If it's client only operation and the server has no rules to enforce, then what does it mean when a client sets a deleted flag but doesn't clear the content?

Reply:
@davidlehn, I think it simply means that the client is misimplemented. Other clients will have to decide what to do about that -- and the spec should probably say to ignore the data -- but either way, I'm not sure what kind of actionable thing would come out of it. What are your specific concerns?

Reply:
@dlongley 2 concerns/questions from implementing the changes:
1. Now that an update is being performed to the document instead of just deleting it, keyResolver needs to be passed as an argument to be able to encrypt the changes. I was unable to find a workaround at this time. Is that an acceptable breaking change, or is there a different solution?
2. Sequence now increases when a document is deleted, since an update has been performed on it. Is that expected and good, or not?

Reply:
@mandyvenables,

> 1. Now that an update is being performed to the document instead of just deleting it, keyResolver needs to be passed as an argument to be able to encrypt the changes. I was unable to find a workaround at this time. Is that an acceptable breaking change, or is there a different solution?

`keyResolver` needs to be passed to which function? I figured we didn't need to do much here on the server except remove the HTTP DELETE stuff. If you're referring to the client, yeah, I'd expect we'd need a breaking change to the remove API over there so that it would take in whatever parameters it needs to pass off to `update` internally.

> 2. Sequence now increases when a document is deleted, since an update has been performed on it. Is that expected and good, or not?

Yes, that's both expected and good.

Reply:
@dlongley Apologies for the confusion, I was referring to adding `keyResolver` to the client, not the server. That has been added [here](https://github.com/digitalbazaar/edv-client/pull/71).

Reply:
@dlongley It does seem unfortunate to move away from method verbs.  REST folks may think a "delete" operation really deletes something so that you would get a 404 if you retrieve it later.  With this system do you get a 200 OK when you ask for a "deleted" resource?  Moving logic into clients means that now all clients need to have validation logic too.  And rules to handle data from other clients that were misimplemented.  And a test suite for all that.  I think I'm unclear on the semantics of this operation, so it's hard to tell if this is the right solution.